constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionLayout.kt

```diff
@@ -748,16 +749,17 @@ fun Transition(@Language("json5") content: String): androidx.constraintlayout.co
                 object : androidx.constraintlayout.compose.Transition {
                     val pixelDp = CorePixelDp { dpValue -> dpValue * dpToPixel }
 
-                    // TODO: there's probably a better way to do this, since there's no reason to
-                    //  call `applyTo` more than once
-                    var applied = false
+                    override fun applyAllTo(transition: Transition, type: Int) {
+                        try {
+                            TransitionParser.parse(parsed, transition, pixelDp)
+                        } catch (e: CLParsingException) {
+                            Log.e("CML", "Error parsing JSON $e")
+                        }
+                    }
 
-                    override fun applyTo(transition: Transition, type: Int) {
+                    override fun applyKeyFramesTo(transition: Transition) {
                         try {
-                            if (!applied) {
-                                TransitionParser.parse(parsed, transition, pixelDp)
-                                applied = true
-                            }
+                            TransitionParser.parseKeyFrames(parsed, transition)
                         } catch (e: CLParsingException) {
```

Comment:
Generally keep the API surface narrow. 
A better approach is to add a flag to the parse to say KeyFrames. only